### PR TITLE
chore: update ts & eslint

### PR DESCRIPTION
**Pull request by rjchow**

### CODE REVIEW

File: src/common/hooks/useKeyPress.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { useState, useEffect } from "react";
-
-export const useKeyPress = (targetKey: any) => {
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
```
Comment (rjchow) [inline src/common/hooks/useKeyPress.ts:2]: think you don't need this disable line after adding the return type below?

File: src/common/utils/document.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { WrappedDocument, getData, v2 } from "@govtechsg/open-attestation";
 
-export const getDocumentId = (document: WrappedDocument) => `0x${document.signature.targetHash}`;
-export const getTokenRegistryAddress = (document: WrappedDocument<v2.OpenAttestationDocument>) =>
+export const getDocumentId = (document: WrappedDocument): string => `0x${document.signature.targetHash}`;
```
Comment (rjchow) [inline src/common/utils/document.ts:4]:
```suggestion
// TODO: change this to use the open-attestation library exported utilities when we upgrade the version
export const getDocumentId = (document: WrappedDocument): string => `0x${document.signature.targetHash}`;
```

File: src/services/verify/index.tsx [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
+  VerificationFragment,
+} from "@govtechsg/oa-verify";
 import { NETWORK_NAME } from "../../config";
 
 const customVerify = verificationBuilder([...openAttestationVerifiers, openAttestationDidIdentityProof]);
 
 export const verifyDocument = async (
   document: WrappedDocument<v3.OpenAttestationDocument> | WrappedDocument<v2.OpenAttestationDocument>
-) => {
+): Promise<VerificationFragment<any>[]> => {
```
Comment (rjchow) [inline src/services/verify/index.tsx:15]:
```suggestion
// TODO: fix types when upgrading oa-verify
): Promise<VerificationFragment<any>[]> => {
```

**Review by rjchow — COMMENTED** [review]
ok mostly lgtm just pending manual testing

File: src/common/hooks/useTitleEscrowContract.ts [hunk 4]
```diff
@@ -4,11 +4,17 @@ import { TitleEscrowFactory } from "@govtechsg/token-registry";
 import { TitleEscrow } from "@govtechsg/token-registry/types/TitleEscrow";
 import { TradeTrustErc721 } from "@govtechsg/token-registry/types/TradeTrustErc721";
 
+interface useTitleEscrowContractProps {
+  titleEscrow: TitleEscrow | undefined;
+  documentOwner: string | undefined;
```
Comment (isaackps) [inline src/common/hooks/useTitleEscrowContract.ts]: changed to ?

File: src/common/hooks/useTokenRegistryContract.ts [hunk 5]
```diff
@@ -3,7 +3,12 @@ import { TradeTrustErc721Factory } from "@govtechsg/token-registry";
 import { TradeTrustErc721 } from "@govtechsg/token-registry/types/TradeTrustErc721";
 import { providers, Signer } from "ethers";
 
-export const useTokenRegistryContract = (address?: string, provider?: providers.Provider | Signer) => {
+export const useTokenRegistryContract = (
+  address?: string,
+  provider?: providers.Provider | Signer
+): {
+  tokenRegistry: TradeTrustErc721 | undefined;
```
Comment (isaackps) [inline src/common/hooks/useTokenRegistryContract.ts]: change to ?

File: src/common/hooks/useKeyPress.ts (hunk 1 above)
Comment (isaackps) [reply src/common/hooks/useKeyPress.ts:2]: need this, its for the "targetKey:any"

File: src/common/utils/document.ts (hunk 2 above)
Comment (isaackps) [reply src/common/utils/document.ts:4]: ok

File: src/common/hooks/useTitleEscrowContract.ts (hunk 4 above)
Comment (isaackps) [reply src/common/hooks/useTitleEscrowContract.ts]: ok

File: src/common/hooks/useTokenRegistryContract.ts (hunk 5 above)
Comment (isaackps) [reply src/common/hooks/useTokenRegistryContract.ts]: ok

**Review by simboonlong — APPROVED** [review]
lgtm

**Pull request merged**